src/core/server/saved_objects/types.ts

```diff
@@ -245,9 +245,9 @@ export interface SavedObjectsType {
    */
   mappings: SavedObjectsTypeMappingDefinition;
   /**
-   * An optional map of {@link SavedObjectMigrationFn | migrations} to be used to migrate the type.
+   * An optional object of {@link SavedObjectMigrationFn | migrations} or a function returning a map of {@link SavedObjectMigrationFn | migrations} to be used to migrate the type.
```

Comment:
nit: use "map" for both the static and return of the function for consistency
```suggestion
   * An optional map of {@link SavedObjectMigrationFn | migrations} or a function returning a map of {@link SavedObjectMigrationFn | migrations} to be used to migrate the type.
```